From: Peter Kokot <peterkokot@gmail.com>
Subject: Fix tests

- ext/zip
  When PHP is compiled and built against libzip that doesn't have AES
  support enabled (if all encryption options are set to OFF in CMake),
  ZipArchive::EM_AES_256 and ZipArchive::EM_AES_192 aren't supported.
  ZipArchive::registerProgressCallback() is only available when using
  libzip 1.3.0 or newer.
  ZipArchive::RDONLY is available only with libzip >= 1.0.0

- ext/standard
  The tempnam() needs some existing directory, otherwise it emits a
  notice (e.g., when running tests from out-of-source build and running
  run-tests.php script from a directory not in the php-src).
---
 ext/standard/tests/file/002.phpt        | 2 +-
 ext/zip/tests/gh18431.phpt              | 4 ++++
 ext/zip/tests/oo_addemptydir_error.phpt | 2 +-
 ext/zip/tests/oo_addglob2.phpt          | 6 ++++++
 ext/zip/tests/oo_encryption.phpt        | 6 ++++++
 5 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/ext/standard/tests/file/002.phpt b/ext/standard/tests/file/002.phpt
index 134f5416fc4..0eb882dc397 100644
--- a/ext/standard/tests/file/002.phpt
+++ b/ext/standard/tests/file/002.phpt
@@ -22,7 +22,7 @@
 blah blah blah blah blah blah blah
 EOD;
 
-$name = tempnam("./ext/standard/tests/file/", "php");
+$name = tempnam(__DIR__, "php");
 $fp = fopen($name, "w");
 fwrite($fp, $data);
 fclose($fp);
diff --git a/ext/zip/tests/gh18431.phpt b/ext/zip/tests/gh18431.phpt
index d4eb89c36c6..dcd7db6f400 100644
--- a/ext/zip/tests/gh18431.phpt
+++ b/ext/zip/tests/gh18431.phpt
@@ -2,6 +2,10 @@
 GH-18431 (Registering ZIP progress callback twice doesn't work)
 --EXTENSIONS--
 zip
+--SKIPIF--
+<?php
+if (!method_exists('ZipArchive', 'registerProgressCallback')) die('skip libzip too old');
+?>
 --FILE--
 <?php
 $file = __DIR__ . '/gh18431.zip';
diff --git a/ext/zip/tests/oo_addemptydir_error.phpt b/ext/zip/tests/oo_addemptydir_error.phpt
index 595c7d9ea23..bd7507a51e3 100644
--- a/ext/zip/tests/oo_addemptydir_error.phpt
+++ b/ext/zip/tests/oo_addemptydir_error.phpt
@@ -4,7 +4,7 @@
 zip
 --SKIPIF--
 <?php
-/* $Id$ */
+if (version_compare(ZipArchive::LIBZIP_VERSION, '1.0.0', '<')) die('skip libzip < 1.0.0');
 ?>
 --FILE--
 <?php
diff --git a/ext/zip/tests/oo_addglob2.phpt b/ext/zip/tests/oo_addglob2.phpt
index 517c0b7fd7f..1432d704de5 100644
--- a/ext/zip/tests/oo_addglob2.phpt
+++ b/ext/zip/tests/oo_addglob2.phpt
@@ -6,6 +6,12 @@
 <?php
 if (!method_exists('ZipArchive', 'setEncryptionName')) die('skip encrytion not supported');
 if(!defined("GLOB_BRACE")) die ('skip requires GLOB_BRACE');
+if (
+    method_exists('ZipArchive', 'isEncryptionMethodSupported')
+    && !ZipArchive::isEncryptionMethodSupported(ZipArchive::EM_AES_256)
+) {
+    die('skip EM_AES_256 encryption not supported');
+}
 ?>
 --FILE--
 <?php
diff --git a/ext/zip/tests/oo_encryption.phpt b/ext/zip/tests/oo_encryption.phpt
index f5207e30759..ccc0a04b8f1 100644
--- a/ext/zip/tests/oo_encryption.phpt
+++ b/ext/zip/tests/oo_encryption.phpt
@@ -5,6 +5,12 @@
 --SKIPIF--
 <?php
 if (!method_exists('ZipArchive', 'setEncryptionName')) die('skip encryption not supported');
+if (
+    method_exists('ZipArchive', 'isEncryptionMethodSupported')
+    && !ZipArchive::isEncryptionMethodSupported(ZipArchive::EM_AES_256)
+) {
+    die('skip EM_AES_256 encryption not supported');
+}
 ?>
 --FILE--
 <?php
